### PR TITLE
Add an optional iconText prop to FoldableCard

### PR DESCRIPTION
**Pull request by allendav**
This will allow the screen-reader-text for the expand-chevron to be specified by the containing component through the `iconText` prop.

It also tests for translate mixin before using it for default text in the event the FoldableCard component is being used apart from `base`

cc @jeffstieler @jkudish

### CODE REVIEW

**Comment by jeffstieler** [comment]
LGTM. :+1:

File: client/components/foldable-card/index.jsx [hunk 1]
```diff
@@ -86,9 +87,11 @@ var FoldableCard = React.createClass( {
 		}
 		if ( this.props.children ) {
 			let iconSize = 24;
+			let iconDefaultText = ( typeof this.translate === 'function' ) ? this.translate( 'More' ) : '';
+			let iconText = this.props.iconText || iconDefaultText;
 			return (
 				<button disabled={ this.props.disabled } className="foldable-card__action foldable-card__expand" onClick={ clickAction }>
-					<span className="screen-reader-text">{ this.translate( 'More' ) }</span>
+					<span className="screen-reader-text">{ iconText }</span>
```
Comment (mtias) [inline client/components/foldable-card/index.jsx]: Should this be called `screenReaderText` or similar instead?

Comment (allendav) [inline client/components/foldable-card/index.jsx]: Sure!  I will change the prop as well

Comment (allendav) [inline client/components/foldable-card/index.jsx]: Fixed in https://github.com/Automattic/wp-calypso/pull/4297/commits/75a286085c8e2adc9086b148f6b47adaba594da2

**Comment by allendav** [comment]
@mtias - good to go?

File: client/components/foldable-card/index.jsx [hunk 2]
```diff
@@ -86,9 +87,11 @@ var FoldableCard = React.createClass( {
 		}
 		if ( this.props.children ) {
 			let iconSize = 24;
+			let screenReaderDefaultText = ( typeof this.translate === 'function' ) ? this.translate( 'More' ) : '';
```
Comment (jkudish) [inline client/components/foldable-card/index.jsx]: since these two variables aren't redefined, why not use `const`?

Comment (allendav) [inline client/components/foldable-card/index.jsx]: good point.  fixing

Comment (allendav) [inline client/components/foldable-card/index.jsx]: @jkudish fixed in https://github.com/Automattic/wp-calypso/pull/4297/commits/868dd9aedc8799acc3c77c8bb803cdd4072557b8

**Comment by jkudish** [comment]
Tests failed, but I think it's unrelated so I re-queued them.

**Comment by allendav** [comment]
Tests are still failing for what appears to be an unrelated area, so I have rebased to see if that clears this up.

**Comment by allendav** [comment]
@mtias - restarting the tests did the trick.  could we please merge this now?  thx!

**Comment by allendav** [comment]
Or, in lieu of @mtias , would you mind reviewing/greenlighting this @scruffian ?

File: client/components/foldable-card/index.jsx [hunk 3]
```diff
@@ -85,10 +86,12 @@ var FoldableCard = React.createClass( {
 			);
 		}
 		if ( this.props.children ) {
-			let iconSize = 24;
+			const iconSize = 24;
+			const screenReaderDefaultText = ( typeof this.translate === 'function' ) ? this.translate( 'More' ) : '';
```
Comment (scruffian) [inline client/components/foldable-card/index.jsx]: Why would `this.translate` not be defined?

**Comment by scruffian** [comment]
Don't forget to squash these commits...

File: client/components/foldable-card/index.jsx (hunk 3 above)
Comment (allendav) [inline client/components/foldable-card/index.jsx]: `this.translate` is not defined when calypso components like FoldableCard are used in a context (e.g. another project) where https://github.com/Automattic/wp-calypso/tree/master/client/boot is not wanted

**Comment by allendav** [comment]
Squashed

File: client/components/foldable-card/index.jsx (hunk 3 above)
Comment (scruffian) [inline client/components/foldable-card/index.jsx]: Would it be possible to abstract this somehow? Maybe this text should always be passed in the props? My concern is that this approach makes things kind of messy.

Comment (allendav) [inline client/components/foldable-card/index.jsx]: I'd like that very much.  I will update this PR to always pass the screen reader text as a prop, including fixing the existing callers to do so.

**Comment by allendav** [comment]
@scruffian - ok, now more generic, and I've updated callers (but only those with header indicator not set to false).

To test:

Navigate to each of the following routes in calypso and ensure that the FoldableCards on those pages have a span.screen-reader-text with “More” (to access the screenreader text, you will have to use Javascript Developer Tools and inspect the chevron on each FoldableCard

http://calypso.localhost:3000/devdocs/design/foldable-card
http://calypso.localhost:3000/me/security/connected-applications
http://calypso.localhost:3000/settings/export/testtesttest.wordpress.com (replace with a blog of your own)
http://calypso.localhost:3000/sharing/testtesttest.wordpress.com (replace with a blog of your own)
http://calypso.localhost:3000/following/edit

Also, launch the new post editor (pencil in the upper right).  Click on the @ symbol in the editor toolbar to add a contact form.  When the contact form dialog appears, inspect the pencil for any of the fields and ensure that the FoldableCards on that dialog have a span.screen-reader-text with “Edit”

**Comment by mtias** [comment]
This is looking good, thanks. Adding "editor" label so we get a few eyes on that section.

**Comment by aduth** [comment]
>Adding "editor" label so we get a few eyes on that section.

Are we using this component anywhere in the editor?

**Comment by mtias** [comment]
Ah, my bad, only on contact form.

**Comment by allendav** [comment]
Yep - contact form only.  OK to merge, @mtias ?

File: client/components/foldable-card/index.jsx [hunk 4]
```diff
@@ -85,10 +87,10 @@ var FoldableCard = React.createClass( {
 			);
 		}
 		if ( this.props.children ) {
-			let iconSize = 24;
+			const iconSize = 24;
 			return (
 				<button disabled={ this.props.disabled } className="foldable-card__action foldable-card__expand" onClick={ clickAction }>
-					<span className="screen-reader-text">{ this.translate( 'More' ) }</span>
+					<span className="screen-reader-text">{ this.props.screenReaderText }</span>
```
Comment (mtias) [inline client/components/foldable-card/index.jsx]: Should we just do `this.props.screenReaderText || this.translate( 'More' )` so it's only needed as a prop when you want to change it?

**Comment by mtias** [comment]
Added a comment, seems we shouldn't have to force people to set this attribute to use the component.

File: client/components/foldable-card/index.jsx (hunk 4 above)
Comment (mtias) [inline client/components/foldable-card/index.jsx]: I see this was already discussed. Mmm...

**Comment by allendav** [comment]
@mtias - please take a look now.  I've made it so that if a screenReaderText prop isn't passed, this.translate gets called.

**Comment by allendav** [comment]
@mtias or @scruffian - is there anything else needed for this PR? I would really like to merge this today.  It has been open a very, very long time.  Can I get a LGTM from either of you?

**Comment by jkudish** [comment]
Let's merge it in.

**Pull request merged**